great_expectations/core/expectation_validation_result.py

```diff
@@ -156,6 +157,31 @@ def __eq__(self, other):  # type: ignore[explicit-override] # FIXME
             # if invalid comparisons are attempted, the objects are not equal.
             return False
 
+    @override
+    def __hash__(self) -> int:
```

Comment:
Maybe add a note that even if 2 objects are equal they might not hash to the same thing. We only compare common result keys in the equality check.